javaslang-benchmark/src/test/java/javaslang/idiom/TupleBenchmark.java

```diff
@@ -0,0 +1,63 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.collection.Array;
+import javaslang.collection.Iterator;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import static javaslang.API.For;
+
+/**
+ * Benchmark for Tuple2 vs an array.
+ */
+public class TupleBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            Tuple2Benchmark.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+
+    @State(Scope.Benchmark)
+    public static class Base {
+        String string;
+
+        @Setup
+        public void setup() {
+            byte[] bytes = new byte[100];
+            new Random().nextBytes(bytes);
+            string = new String(bytes);
+        }
+    }
+
+    public static class Tuple2Benchmark extends Base {
+
+        @Benchmark
+        public void javaslang_tuple2_creation(Blackhole bh) {
```

Comment:
Sorry for mistakes, will fix all aforementioned issues. 
I was on vacation, so didn't have much time to work on this, but hopefully will fix the PR today.

Thanks again everyone for your patience.